models/buffer_disk.go

```diff
@@ -272,7 +267,18 @@ func (b *DiskBuffer) Stats() BufferStats {
 }
 
 func (b *DiskBuffer) Close() error {
-	return b.file.Close()
+	if err := b.file.Close(); err != nil {
+		return fmt.Errorf("closing buffer failed: %w", err)
+	}
+
+	// Remove all remaining data on disk to make sure we won't get any metric
+	// in cases where the buffer is empty. This is required because we cannot
+	// truncate all metrics from the buffer.
+	if b.isEmpty {
+		return os.RemoveAll(b.path)
+	}
+
+	return nil
```

Comment:
Reading b.isEmpty without acquiring the mutex could cause a race condition if other goroutines are still accessing the buffer. 
May I suggest:

```suggestion
        b.Lock()
	isEmpty := b.isEmpty
	b.Unlock()
	
	if err := b.file.Close(); err != nil {
		return fmt.Errorf("closing buffer failed: %w", err)
	}

	// Remove all remaining data on disk to make sure we won't get any metric
	// in cases where the buffer is empty. This is required because we cannot
	// truncate all metrics from the buffer.
	if isEmpty {
		return os.RemoveAll(b.path)
	}

	return nil
```

Reply:
Good point! Fixed here.